Added Open Class Private Data Feature For Edit Time

shivamanikandan opened the pull request:
@ItsMeAshiq Please test, review and merge the PR

shivamanikandan committed:
Added Right Click Plugin For Typedef Constants

Committing all the created VIs
A test vi is added (will be deleted at the end of dev)

shivamanikandan committed:
Feature added to show RCM in constants & terminals

Made Execute VI to work fine with typedef

shivamanikandan committed:
Code changes done for  constants, controls & indicators in run time

Code changes are done for Opening TypeDef. and Class private data from block diagram constants, controls & indicators in run time

shivamanikandan committed:
Added Build Qualified Class Private Data Name.vi

To resolve the conflict with Build Qualified Class Private Data Name.vi, moved it to open-typedef-class.llb folder and committed

shivamanikandan committed:
Cleaned up the code base

Removed redundant files
Moved VIs, Ctls to src\tests which are used only for testing
Created custom class to avoid the dependency with VI packages
Added all the files related open-typedef-class.llb to Project.lvproj itself

ItsMeAshiq committed:
Rename the Plugin

Rename the plugin to `OpenTypeDef-RunTime`

ItsMeAshiq committed:
Reorganize Tests

Move Tests into more suitable folder
Improve Test Main code

shivamanikandan committed:
Merge branch 'LabVIEW-Nation:main' into main

shivamanikandan committed:
Merge branch 'LabVIEW-Nation:main' into main

shivamanikandan committed:
Added Open Class Private Data Feature In Edit Time

Created new right click plugin
Updated VIs to open class private data in edit time

ItsMeAshiq commented on src/code/open-class-private-data-edit-time.llb/open-class-private-data-edit-time.vi:
<img width="218" height="343" alt="image" src="https://github.com/user-attachments/assets/9c1352e8-0d93-4904-a514-597e6831eff8" />

We need to move the RCM to show immediately below to the entry **Show Class Library**

Can refer how to get the ID of **Show Class Library** entry from this VI: https://github.com/LabVIEW-Nation/Show-Attached-Object--Menu-for-BD-Comments/blob/main/src/code/Show-Attached-Object--Menu-for-BD-Comments.llb/Show-Attached-Object--Menu-for-BD-Comments.vi
<img width="1435" height="885" alt="image" src="https://github.com/user-attachments/assets/7bc9a9ae-808a-441b-bb05-3dc05a504c4e" />

ItsMeAshiq replied:
<img width="218" height="353" alt="image" src="https://github.com/user-attachments/assets/7ea5ee84-3ffd-4f61-a008-a831dd2757a6" />
Nice 👍

ItsMeAshiq commented on src/code/open-class-private-data-edit-time.llb/open-class-private-data-edit-time.ctl:
This toolkits behavior is expecting some RCM location mapping as I mentioned in the other comment.
So, I feel it would be easier to manage if this plugin is an independent one.

So, I propose following changes,
1. I suggest to rename this plugin to `OpenClassPrivateData-EditTime`
2. We can update the affected items to work for Class specific items,
   * <img width="212" height="150" alt="image" src="https://github.com/user-attachments/assets/2acf9e36-8ba1-44ca-85dc-12e65ef48600" />
   * <img width="166" height="593" alt="image" src="https://github.com/user-attachments/assets/ab46abb5-878b-4114-9bcf-e1ff6377811c" />
   * <img width="174" height="289" alt="image" src="https://github.com/user-attachments/assets/f6db6d38-c1f9-46f4-a725-de1dc962907d" />

shivamanikandan committed:
Merge branch 'LabVIEW-Nation:main' into main

shivamanikandan committed:
Updated OpenTypedef Run Time

Modified code to make it work for typdefs with same name but in diff lib, class or location

shivamanikandan committed:
Code changes to reposition the menu

Changed folder name to OpenClassPrivateData-EditTime
Changed affected items cluster to class control and constant refs
Re-positioned the open class private data menu after "Show Class Libarary" menu

shivamanikandan commented:
@ItsMeAshiq I have done the following,

1. Changed folder name to OpenClassPrivateData-EditTime
2. Changed affected items cluster to class control and constant refs
3. Re-positioned the open class private data menu after "Show Class Libarary" menu

Please test, review and LMK if you have any feedback else merge it and resume the integration task

ItsMeAshiq committed:
Save Files to LV 20

Test & save files to LV 20
Add OpenClassPrivateData-EditTime folder to the project
